packages/core/http/core-http-router-server-mocks/src/versioned_router.mock.ts

```diff
@@ -6,19 +6,77 @@
  * Side Public License, v 1.
  */
 
-import type { VersionedRouter, VersionedRoute } from '@kbn/core-http-server';
+import type {
+  VersionedRouter,
+  VersionedRoute,
+  VersionedRouteConfig,
+  AddVersionOpts,
+} from '@kbn/core-http-server';
 
-const createMockVersionedRoute = (): VersionedRoute => {
-  const api: VersionedRoute = { addVersion: jest.fn(() => api) };
+export type MockedVersionedRoute = jest.Mocked<VersionedRoute>;
+
+const createMockVersionedRoute = (): MockedVersionedRoute => {
+  const addVersion = jest.fn((_, __) => api);
+  const api: MockedVersionedRoute = { addVersion };
   return api;
 };
 
-export type MockedVersionedRouter = jest.Mocked<VersionedRouter<any>>;
+export type MockedVersionedRouter = jest.Mocked<VersionedRouter<any>> & {
+  getRoute: (method: keyof VersionedRouter, path: string) => RegisteredVersionedRoute;
+};
+
+const createMethodHandler = () => jest.fn((_) => createMockVersionedRoute());
+
+export const createVersionedRouterMock = (): MockedVersionedRouter => {
+  const router: Omit<MockedVersionedRouter, 'getRoute'> = {
+    delete: createMethodHandler(),
+    get: createMethodHandler(),
+    patch: createMethodHandler(),
+    post: createMethodHandler(),
+    put: createMethodHandler(),
+  };
+
+  return {
+    ...router,
+    getRoute: getRoute.bind(null, router),
+  };
+};
+
+export interface RegisteredVersionedRoute {
+  config: VersionedRouteConfig<any>;
+  versions: { [version: string]: { config: AddVersionOpts<any, any, any>; handler: any } };
```

Comment:
should `handler` here be:

```
RequestHandler<any, any, any, any, any, KibanaResponseFactory>
```

This will ensure that what we get back matches what was registered - with the exception that `Context` is being set to `any`.

Reply:
Yes! Good catch, I left `any` there while I was concentrating on something else and did not return to it! I'll update per your suggestion.